ui/apps/platform/cypress/integration/certexpiration.test.js

```diff
@@ -1,125 +1,175 @@
 import dateFns from 'date-fns';
-import withAuth from '../helpers/basicAuth';
+
 import * as api from '../constants/apiEndpoints';
 import { selectors } from '../constants/CertExpiration';
-import { visitMainDashboard } from '../helpers/main';
+import { systemConfigUrl } from '../constants/SystemConfigPage';
+import withAuth from '../helpers/basicAuth';
+import {
+    credentialexpiryCentralAlias,
+    credentialexpiryScannerAlias,
+    mypermissionsAlias,
+    visitWithGenericResponses,
+} from '../helpers/visit';
+
+/*
+ * Visit System Configuration page for certificate expiration tests:
+ * It has a minimal number of page-specific requests.
+ * The credentialexpiry and mypermissions requests are generic for any page.
+ */
+function visitForCredentialExpiry(staticResponseMapGeneric) {
+    visitWithGenericResponses(systemConfigUrl, staticResponseMapGeneric, {
+        config: {
+            method: 'GET',
+            url: api.system.config,
+        },
+    });
+}
 
 describe('Cert Expiration Banner', () => {
     withAuth();
 
-    const mockCertExpiryAndVisitHomepage = (endpoint, expiry) => {
-        cy.intercept('GET', endpoint, {
-            body: { expiry },
-        }).as('certExpiry');
-        visitMainDashboard();
-        cy.wait('@certExpiry');
-    };
-
     describe('Central', () => {
         it('should not display banner if central cert is expiring more than 14 days later', () => {
             const expiry = dateFns.addHours(dateFns.addDays(new Date(), 15), 1);
-            mockCertExpiryAndVisitHomepage(api.certExpiry.central, expiry);
+            const staticResponseMapGeneric = {
+                [credentialexpiryCentralAlias]: {
+                    body: { expiry },
+                },
+            };
+            visitForCredentialExpiry(staticResponseMapGeneric);
```

Comment:
The naming for this function seems a bit different than the others. I think it's also a bit hard to understand what it's doing just by the name. If I had to define the purpose of this function I would say "visit the main dashboard page with credential expiry mocked". We have a number of options that would be descriptive enough. One could be
`visitMainDashboardWithCredentialExpiryResponse`